### ISSUE TITLE
Latest nightly release doesn't support `v:lua` in `operatorfunc`

### ISSUE THREAD

**Issue by echasnovski** [neovim/neovim#19857]
### Neovim version (nvim -v)

NVIM v0.8.0-dev-896-g1a57cd383

### Vim (not Nvim) behaves the same?

Vim doesn't have `v:lua`

### Operating system/version

EndeavourOS 5.19.1-arch2-1

### Terminal name/version

xfce4-terminal 1.0.4 (Xfce 4.16)

### $TERM environment variable

xterm-256color

### Installation

appimage

### How to reproduce the issue

1. Create initialization file (for example, 'init-operatorfunc-lua.lua') with the following content:
```lua
_G.my_operatorfunc = function() _G.inside_operatorfunc = true end

_G.operator = function()
  vim.cmd('set operatorfunc=v:lua._G.my_operatorfunc')
  return 'g@'
end

vim.keymap.set('n', 'go', 'v:lua._G.operator()', { expr = true })
```
2. Open Neovim with `nvim -u init-operatorfunc-lua.lua init-operatorfunc-lua.lua`.
3. Type `goiw`.

### Expected behavior

Executes without error `E117: Unknown function: v:lua._G.my_operatorfunc`.

### Actual behavior

https://user-images.githubusercontent.com/24854248/185742731-a0a5b22b-7228-44c0-8748-69d419c96688.mp4

Everything executes as expected in NVIM v0.8.0-dev-648-gd15a66d80 . I suspect that e8618df7f826d2ca4d524f12fc712d7c52ea158c has something to do with this.

**Comment by zeertzjq** [comment]
Fixed by #19855

**Comment by echasnovski** [comment]
Oh, sorry for the noise then. It didn't come up when I searched for `operatorfunc`.
